lib/SPIRV/libSPIRV/spirv.hpp

```diff
@@ -71,21 +71,18 @@ enum ExecutionModel {
     ExecutionModelFragment = 4,
     ExecutionModelGLCompute = 5,
     ExecutionModelKernel = 6,
-    ExecutionModelCount, /* internal use only */
```

Comment:
I got response that adding *Count at the end of enum definition is not desirable since an extension may add other enum values which renders *Count not a generally useful solution.

I think that makes sense. Looking back at this implementation

    inline bool
    isValid(SPIRVExecutionModelKind E) {
        return (unsigned)E <= (unsigned)ExecutionModelKernel;
    }

It may not be as bad as we thought, because this _is_ what we currently know valid enum values. Let's say if some new enum values are added and we don't know about them, chances are that our translator is not able to handle them, so these newly added values _are_ invalid values for us. So we will get assertion, and we look at them and handle them.

So I think we should be OK to remove the internally used *Count enums and change invalid() to the new implementation.